src/main/java/org/folio/service/pieces/ItemRecreateInventoryService.java

```diff
@@ -0,0 +1,84 @@
+package org.folio.service.pieces;
+
+import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS;
+import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS_NAME;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.models.ItemStatus;
+import org.folio.models.pieces.PieceUpdateHolder;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.service.inventory.InventoryItemManager;
+import org.folio.service.pieces.flows.DefaultPieceFlowsValidator;
+
+public class ItemRecreateInventoryService {
+
+  protected static final int ITEM_QUANTITY = 1;
+  private static final Logger logger = LogManager.getLogger(ItemRecreateInventoryService.class);
+
+  private final InventoryItemManager inventoryItemManager;
+
+  public ItemRecreateInventoryService(InventoryItemManager inventoryItemManager) {
+    this.inventoryItemManager = inventoryItemManager;
+  }
+
+  // Return an Id of the recreated Item
+  public Future<String> recreateItemInDestinationTenant(PieceUpdateHolder holder, RequestContext srcLocCtx, RequestContext dstLocCtx) {
+    // Example Case: Member Tenant 1 (University) -> Member Tenant 2 (College)
+    // Create Item in Member Tenant 2 with the same Item Id
+    // Delete Item in Member Tenant 1 by Item Id
+    var itemFuture = Promise.<String>promise();
+    var piece = holder.getPieceToUpdate();
+    var compPol = holder.getPoLineToSave();
+
+    try {
+      logger.debug("Handling {} items for PO Line and holdings with id={}", ITEM_QUANTITY, piece.getHoldingId());
+
+      if (piece.getFormat() == Piece.Format.ELECTRONIC && DefaultPieceFlowsValidator.isCreateItemForElectronicPiecePossible(piece, compPol)) {
+        inventoryItemManager.recreateMissingElectronicItems(compPol, piece, ITEM_QUANTITY, dstLocCtx)
+          .onSuccess(idS -> itemFuture.complete(idS.get(0)))
+          .onFailure(itemFuture::fail);
+      } else if (DefaultPieceFlowsValidator.isCreateItemForNonElectronicPiecePossible(piece, compPol)) {
```

Comment:
similar if else statement as in PieceUpdateInventoryService, that can be potentially extracted in common class that will have 2 abstact method to override, but can be looked as overengineering. I am ok to leave as it, just thoughts